examples/src/main/java/io/grpc/examples/header/HeaderClientInterceptor.java

```diff
@@ -0,0 +1,37 @@
+package io.grpc.examples.header;
+
+import io.grpc.*;
+
+import java.util.logging.Logger;
+
+/**
+ * @author zhaohaifeng
+ * @since 2015-03-16
+ */
+public class HeaderClientInterceptor implements ClientInterceptor {
```

Comment:
https://github.com/grpc/grpc-java/blob/master/stub/src/main/java/io/grpc/stub/MetadataUtils.java#L71 is useful for client-side. I don't think we have something already for server-side.